src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -89,6 +89,395 @@ public class ConfigurationTests
+              ""data-source"": {
+                ""database-type"": ""mssql"",
+                ""options"": {
+                  ""set-session-context"": true
+                }
+              },
+              ""runtime"": {
+                ""rest"": {
+                  ""enabled"": true,
+                  ""path"": ""/api""
+                },
+                ""graphql"": {
+                  ""enabled"": true,
+                  ""path"": ""/graphql"",
+                  ""allow-introspection"": true
+                },
+                ""host"": {
+                  ""cors"": {
+                    ""origins"": [""http://localhost:5000""],
+                    ""allow-credentials"": false
+                  },
+                  ""authentication"": {
+                    ""provider"": ""StaticWebApps"",
+                    ""jwt"": {
+                      ""audience"": null,
+                      ""issuer"": null
+                    }
+                  },
+                  ""mode"": ""development""
+                }
+              },
+              ""entities"": {
+                ""GetBooks"": {
+                  ""source"": {
+                    ""object"": ""get_books"",
+                    ""type"": ""stored-procedure"",
+                    ""parameters"": null,
+                    ""key-fields"": null
+                  },
+                  ""graphql"": {
+                    ""enabled"": true,
+                    ""operation"": ""query"",
+                    ""type"": {
+                      ""singular"": ""GetBooks"",
+                      ""plural"": ""GetBooks""
+                    }
+                  },
+                  ""rest"":{
+                    ""methods"": [
+                      ""get""
+                    ]
+                  },
+                  ""permissions"": [
+                    {
+                      ""role"": ""anonymous"",
+                      ""actions"": [
+                        {
+                          ""action"": ""execute"",
+                          ""fields"": null,
+                          ""policy"": {
+                            ""request"": null,
+                            ""database"": null
+                          }
+                        }
+                      ]
+                    }
+                  ],
+                  ""mappings"": null,
+                  ""relationships"": null
+                }
+              }
+        }";
+
+        /// <summary>
+        /// A config file with a SP entity for which REST APIs are disabled.
+        /// This config string is used for validating that none of the REST methods are enabled.
+        /// </summary>
+        public const string SP_CONFIG_WITH_REST_DISABLED = @"
+            {
+              ""$schema"": ""https://github.com/Azure/data-api-builder/releases/download/vmajor.minor.patch/dab.draft.schema.json"",
```

Comment:
Hard coding different variations of the config file for the various test cases doesn't seem scalable. 
Can we do better? Only parts of the config are varying, can we keep a common const string, then build new configs for the parts that are varying?

Reply:
Updated to generate the configs dynamically